### ISSUE TITLE
Show all Users from a given Role

### ISSUE THREAD

**Issue by markusvt** [aspnet/Identity#765]
Hello, i spent the last day to figure out how to do it, but it seems with the given functions of UserManager and RoleManager, the only possible way to achieve this is to enumerate all Users and check if the user has specified role.

Is there a way to do this better than enumerate over all user?

**Comment by blowdart** [comment]
What is your scenario here? An admin panel or dashboard? This has a performance cost. What we could do is give you the user IDs in a role, and then you need to resolve those to a user, using the UserStore.

**Comment by markusvt** [comment]
Yes it is an admin panel in which one can manage the roles and their assigned users.
I just wondered that the functionality does not exist as I could imagine that this scenario appears quite often. I guess a list of UserIds is sufficient.

By the way, in my specific scenario (I use EF7 and a SQL Server) I solved the whole problem by developing a database-join between the User, UserRole and Role Table

**Comment by blowdart** [comment]
Ah good, at least you're not blocked. Yes, we're not aiming identity at admin panel scenarios. We figure that if you're writing an admin panel you can do this for yourself. There are somewhat interesting performance hits in this sort of link, because you can end up loading the entire database.

We'll look at the user ID link.

**Comment by HaoK** [comment]
@blowdart @divega is this something for the backlog or 1.0?

**Comment by blowdart** [comment]
1.0

**Comment by divega** [comment]
:+1:
